### ISSUE TITLE
rotaryWithScroll library doesn't respect use crown haptic settings

### ISSUE THREAD

**Issue by blakadder** [google/horologist#1750]
I am directed by the creators of the Home Assistant Companion app to file a bug report;

On TicWatch Pro 5 with OS 3.5 the app doesn't respect the "Use crown" vibration system settings and keeps vibration on crown use. Furthermore the type of vibrations are not the usual ones used when Use crown in enabled bit instead the strong vibration that's used for touch events is used

**Comment by yschimke** [comment]
Thanks for the report. We can add specific haptics but we would need to know which ones are preferred. I'll see if it's doing something stupid and using the ones for a different model.

Is there documentation for this setting? Or it's a standard wear one?

**Comment by jpelgrom** [comment]
I can confirm the same 'issue' exists on a Samsung Galaxy Watch 4 with the touch bezel (= rotary input on this device). In the system settings app > General > Touch bezel, there is an option "Vibration feedback". Sounds like it is kind of standard Wear but depends on the device what it's called and where it is located.

Turning it off doesn't change the behavior of apps using the `rotaryWithScroll` modifier, while other apps that (presumably) use a classic view do change and stop vibrating when rotating. Easy to spot as thin vs thick scroll bar.

**Comment by blakadder** [comment]
> Thanks for the report. We can add specific haptics but we would need to know which ones are preferred. I'll see if it's doing something stupid and using the ones for a different model.
> 
> Is there documentation for this setting? Or it's a standard wear one?

Specific or not the existing one is overly aggressive and should be toned down.

The setting also exists on the Pixel Watch as "Using crown" - https://www.verizon.com/support/knowledge-base-303381/

**Comment by yschimke** [comment]
So I think we need to do one of two things

1) Find the relevant haptics code for this device, or hopefully class of devices.
2) Fallback to reading the setting, possibly `Settings.Global.getInt(context.contentResolver, "wear_vibrate_for_rotary_input")`

It's possible neither are public APIs.

A quick test of 1, are you able to add this in and test on that device?

```
public class LikePixelWatchRotaryHapticFeedback(private val view: View) : RotaryHapticFeedback {

    @ExperimentalHorologistApi
    override fun performHapticFeedback(
        type: RotaryHapticsType,
    ) {
        when (type) {
            RotaryHapticsType.ScrollItemFocus -> {
                view.performHapticFeedback(
                    if (Build.VERSION.SDK_INT >= 33) {
                        ROTARY_SCROLL_ITEM_FOCUS
                    } else {
                        WEAR_SCROLL_ITEM_FOCUS
                    },
                )
            }

            RotaryHapticsType.ScrollTick -> {
                view.performHapticFeedback(
                    if (Build.VERSION.SDK_INT >= 33) ROTARY_SCROLL_TICK else WEAR_SCROLL_TICK,
                )
            }

            RotaryHapticsType.ScrollLimit -> {
                view.performHapticFeedback(
                    if (Build.VERSION.SDK_INT >= 33) ROTARY_SCROLL_LIMIT else WEAR_SCROLL_LIMIT,
                )
            }
        }
    }

    private companion object {
        // Hidden constants from HapticFeedbackConstants.java specific for Pixel Watch
        // API 33
        public const val ROTARY_SCROLL_TICK: Int = 18
        public const val ROTARY_SCROLL_ITEM_FOCUS: Int = 19
        public const val ROTARY_SCROLL_LIMIT: Int = 20

        // API 30
        public const val WEAR_SCROLL_TICK: Int = 10002
        public const val WEAR_SCROLL_ITEM_FOCUS: Int = 10003
        public const val WEAR_SCROLL_LIMIT: Int = 10003
    }
}
```

**Comment by jpelgrom** [comment]
> A quick test of 1, are you able to add this in and test on that device?

After adding this in, I no longer get any haptic feedback at all even though it is enabled on the Galaxy Watch 4 with Wear OS 4 I'm using.

**Comment by yschimke** [comment]
I think we would stick with the current Samsung code. But mainly interested in devices other than pixel or Samsung.

Is the current published code not working for Samsung for you.

**Comment by Kpeved** [comment]
I created a prototype branch which should use device specific haptics based on the Wear version rather than specific device. 
https://github.com/google/horologist/tree/version_specific_rotary_haptics_16.10
Please test this solution and see whether it solves your problem @blakadder @jpelgrom . 
Unfortunately I don't have a lot of devices at hand for testing so would be great to test that before submitting.

**Comment by jpelgrom** [comment]
Sorry for the delay, I see the change was already merged. I only have a Galaxy Watch 4 to test which is using the same haptics settings as with <= 0.5.8 so nothing to test for me.

**Comment by Kpeved** [comment]
We returned Samsung haptics to its original state, so it should be working as before - haptics setting shouldn't influence compose haptics on GW devices

**Comment by yschimke** [comment]
Released with 0.4.16.  If still happening, raise an issue with specific device details, but should be fixed.
